### PR TITLE
Helm restrict resources

### PR DESCRIPTION
**Pull request by rathberm**
### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: helm/superset/values.yaml [hunk 1]
```diff
@@ -253,6 +253,14 @@ supersetNode:
   podAnnotations: {}
   ## Labels to be added to supersetNode pods
   podLabels: {}
+  # these setting will overwrite Values.resources
```
Comment (danielhass) [inline helm/superset/values.yaml:256]: I would suggest that we extend this comment and align it with the other ones, some along those lines:

`Resource settings for the supersetNode pods - these settings overwrite might existing values from the global resources object defined above.`

This should be altered for all other specific resource settings as well

Comment (danielhass) [reply helm/superset/values.yaml:256]: Furthermore I would also alter the following comment: https://github.com/rathberm/superset/pull/1/files#diff-e5bd590aeb8a54f45bf03f06039525bcf6eb92b24020a1b8634d2fb05cc8ca75L202

It should reflect the changes introduced by you and inform the user that specific settings can be entered below an explain which values take precedence.

**Review by danielhass — CHANGES_REQUESTED** [review]

File: helm/superset/values.yaml (hunk 1 above)
Comment (danielhass) [reply helm/superset/values.yaml:256]: The above link seems to be broken, what I'm referencing is the comment of the global resources setting:
```
resources: {}
  # We usually recommend not to specify default resources and to leave this as a conscious
  # choice for the user. This also increases chances charts run on environments with little
  # resources, such as Minikube. If you do want to specify resources, uncomment the following
  # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
  # limits:
  #   cpu: 100m
  #   memory: 128Mi
  # requests:
  #   cpu: 100m
  #   memory: 128Mi
```

Comment (rathberm) [reply helm/superset/values.yaml]: @danielhass please check out the changes I introduced.

**Pull request closed without merging**